apps/arweave/src/ar_mining_server.erl

```diff
@@ -594,7 +592,17 @@ post_solution(not_set, Solution, State) ->
 			ar:console("WARNING: the solution we found is invalid. Check logs for more "
 					"details~n");
 		{true, PoACache, PoA2Cache} ->
-			ar_events:send(miner, {found_solution, Solution, PoACache, PoA2Cache})
+			ar_events:send(miner, {found_solution, miner, Solution, PoACache, PoA2Cache})
+	end;
+post_solution(ExitPeer, Solution, #state{ is_pool_client = true }) ->
+	case ar_http_iface_client:post_partial_solution(ExitPeer, Solution, false) of
+		{ok, _} ->
+			ok;
+		{error, Reason} ->
+			?LOG_WARNING([{event, found_partial_solution_but_failed_to_reach_exit_node},
+					{reason, io_lib:format("~p", [Reason])}]),
+			ar:console("We found a partial solution but failed to reach the exit node, "
+					"error: ~p.", [io_lib:format("~p", [Reason])])
 	end;
 post_solution(ExitPeer, Solution, _State) ->
 	case ar_http_iface_client:cm_publish_send(ExitPeer, Solution) of
```

Comment:
Do we still need `cm_publish_send`? It'd be great to remove it and just using `post_partial_solution`

Reply:
`cm_publish_send` posts to `/coordinated_mining/publish`. I would not mix coordinated mining interactions with the pool interactions cause they are different use-cases and have slightly different trust requirements. An alternative might be to always send partial solutions directly to the pool from every node, but it complicates configuration a bit.